src/OrleansProviders/Streams/Memory/MemoryMessageBody.cs

```diff
@@ -29,16 +30,28 @@ public interface IMemoryMessageBodySerializer
     /// Default IMemoryMessageBodySerializer
     /// </summary>
     [Serializable]
-    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer
+    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer, IOnDeserialized
```

Comment:
After looking at this a bit more, I think a better solution would be for the [de]serialization calls in the IMemoryMessageBodySerializer to take a serialization manager as an argument, and for the GetEvents call in the IBatchContainer to also take a serialization manager.  The stream consumer extension getting the messages from the batch container would need to pass in a serialization manager.  This, IMO, is a more straight forward approach.